drivers/sensor/sensor_sim/sensor_sim.c

```diff
@@ -203,21 +255,163 @@ static double generate_pseudo_random(void)
 }
 
 /*
- * @brief Calculates sine from uptime
- * The input to the sin() function is limited to avoid overflow issues
+ * @brief Generate value for acceleration signal toggling between two values on fetch.
+ *
+ * @param chan[in]	Selected sensor channel.
+ * @param out_val[out]	Pointer to the variable that is used to store result.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+static int generate_toggle(enum sensor_channel chan, double *out_val)
+{
+	static const double amplitude = 20.0;
+	static double val_sign = 1.0;
+
+	double res_val = amplitude * val_sign;
+
+	*out_val = res_val;
+	if (chan == SENSOR_CHAN_ACCEL_XYZ) {
+		*(out_val + 1) = res_val;
+		*(out_val + 2) = res_val;
+	}
+
+	val_sign *= -1.0;
+
+	return 0;
+}
+
+/*
+ * @brief Calculate sine wave value.
+ *
+ * @param time[in]	Time for generated value (lower than the sine wave period).
+ * @param period[in]	Sine wave period.
+ *
+ * @return Sine wave value for given time.
+ */
+static double sine_val(uint32_t time, uint32_t period)
+{
+	static const double amplitude = 1.0;
+	double angle = 2 * M_PI * time / period;
+
+	return amplitude * sin(angle);
```

Comment:
Is it necessary to multiply with the amplitude when it's always 1.0?

Reply:
Removed